tests/src/test/scala/cats/tests/NonEmptyVectorTests.scala

```diff
@@ -0,0 +1,185 @@
+package cats
+package tests
+
+import cats.kernel.laws.{GroupLaws, OrderLaws}
+
+import cats.data.NonEmptyVector
+import cats.laws.discipline.{ComonadTests, FunctorTests, SemigroupKTests, FoldableTests, MonadTests, SerializableTests, CartesianTests, TraverseTests, ReducibleTests, MonadRecTests}
+import cats.laws.discipline.arbitrary._
+
+class NonEmptyVectorTests extends CatsSuite {
+  // Lots of collections here.. telling ScalaCheck to calm down a bit
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    PropertyCheckConfig(maxSize = 5, minSuccessful = 20)
+
+  checkAll("NonEmptyVector[Int]", OrderLaws[NonEmptyVector[Int]].eqv)
+
+  checkAll("NonEmptyVector[Int] with Option", TraverseTests[NonEmptyVector].traverse[Int, Int, Int, Int, Option, Option])
+  checkAll("Traverse[NonEmptyVector[A]]", SerializableTests.serializable(Traverse[NonEmptyVector]))
+
+  checkAll("NonEmptyVector[Int]", ReducibleTests[NonEmptyVector].reducible[Option, Int, Int])
+  checkAll("Reducible[NonEmptyVector]", SerializableTests.serializable(Reducible[NonEmptyVector]))
+
+  implicit val iso = CartesianTests.Isomorphisms.invariant[NonEmptyVector]
+
+  // Test instances that have more general constraints
+  {
```

Comment:
I don't think we need to add braces for separate scopes for the following tests. That's just done for tests where there's an abstract `F[_]` and we need to test cases where there are different instances available for `F[_]`.